src/lib/components/workspace/members/Tile.svelte

```diff
@@ -0,0 +1,403 @@
+      
+      `}
+    onSuccess={handleMemberDemotePopUpSuccess}
+    onCancel={handleMemberDemotePopUpCancel}
+  />
+{/if}
+
+{#if isMemberOwnershipPopup}
+  <MemberChangeRolePopup
+    auth={true}
+    title={`Changing Role?`}
+    teamName={openTeam?.name}
+    teamLogo={openTeam?.logo}
+    description={`
+      <div class="d-flex tile rounded mb-3">
+    <div
+      class="info d-flex align-items-center"
+    >
+      <div class="d-flex align-items-center justify-content-center" style="width: 36px;
+      border: 1px solid var(--border-color);
+      height: 36px;
+      border-radius: 50%;">
+        <span>${user.name[0].toUpperCase()}</span>
+      </div>
+      <div class="name px-2">
+        <span style="font-size:12px;" class="text-whiteColor">${
+          user.name
+        }</span><br />
+        <span style="font-size:12px;" class="text-textColor">${
+          user.email
+        }</span>
+      </div>
+    </div>
+  </div>
+      
+      <p style="font-size:12px;" class="text-textColor">
+        You are assigning the role of <span class="text-whiteColor">‘Owner’</span> to ${
+          user.name
+        }. All the Owner’s access will be transferred to ${
+          user.name
+        } and you will be demoted to Admin. This action cannot be undone.  </p>
+      
+      `}
+    onSuccess={handleMemberOwnershipPopUpSuccess}
+    onCancel={handleMemberOwnershipPopUpCancel}
+  />
+{/if}
+
+{#if isMemberInfoPopup}
+  <MemberInfoPopup
+    {owner}
+    title={`Access to ${openTeam.name}`}
+    {user}
+    workspaces={workspaces.map((elem) => {
+      let element = elem.toMutableJSON();
+      for (let i = 0; i < element.users.length; i++) {
+        if (element.users[i].id === user.id) {
+          element.position = elem.users[i].role;
+          break;
+        }
+      }
+      return element;
+    })}
+    {userType}
+    {userId}
+    onCancel={handleMemberInfoPopUpCancel}
+    {handleMemberPopUpCancel}
+    {handleMemberPromotePopUpCancel}
+    {handleMemberDemotePopUpCancel}
+    {handleMemberOwnershipPopUpCancel}
+    {teamServiceMethods}
+  />
+{/if}
+<div class="d-flex tile rounded align-items-center">
+  <div
+    class="info d-flex align-items-center"
+    on:click={() => {
+      isMemberInfoPopup = true;
+    }}
+  >
+    <div class="icon d-flex align-items-center justify-content-center">
+      <span>{user.name[0].toUpperCase()}</span>
+    </div>
+    <div class="name px-2">
+      <span style="font-size:12px;" class="text-whiteColor"
+        >{user.name} {owner ? "(You)" : ""}</span
+      ><br />
+      <span style="font-size:12px;" class="text-textColor">{user.email}</span>
+    </div>
+  </div>
+  <div class="position">
+    {#if (userType === TeamRole.OWNER && user.role === TeamRole.MEMBER) || (userType === TeamRole.ADMIN && user.role === TeamRole.MEMBER)}
+      <MemberDropdown
+        id={user.id + uuidv4()}
+        data={[
```

Comment:
Can we break this and take value for data from a function. Maybe you can pass a param and then splice the array accordingly before sending it back.

Reply:
implemented permissions function